---
author: Mrinal Subash
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    dev: 'svg'
  md_document:
    variant: gfm
bibliography: bibliography.ris
nocite: '@*'
---

## Variant Calling

The first step towards Variant Calling is alignment.For alignment, the tools that are being used are BWA(Burrow Wheel Alligner),Speedseq,Bowtie2,BFAST,Illumina Isaak WGS,etc.In this code, we are basically using the BWA aligner.
BWA is a software package for mapping low-divergent sequences against a large reference genome, such as the human genome. It consists of three algorithms: BWA-backtrack, BWA-SW and BWA-MEM.The first algorithm is designed for Illumina sequence reads upto 100bp,while the rest two for longer sequences ranged from 70bp to 1Mbp.

For the second step of getting the reads,we use the fastq-dump function which allows to split files SRR6808334 and gets the specific .log and .err function.

For the third step ,  Somatic variant callingis performed.Somatic variant calling includes steps like SNP , Indels, CNVs and SVs.After getting the reads from SRR6808334, we are trying to trim thte reads using Trimmomatic.Trimmomatic helps to trim the read files and distinguish between the paired and unpaired fastq files. Trimmomatic as a more flexible and efficient preprocessing tool, which could correctly handle paired-end data. The value of NGS read preprocessing is demonstrated for both reference-based and reference-free tasks. Trimmomatic is shown to produce output that is at least competitive with, and in many cases superior to, that produced by other tools, in all scenarios tested .

For the fourth step, the genomes are subsequently indexed. Germline variant callingis performed.A germline variant caller generally has a ploidy-based genotyping algorithm built in to part of the algorithm/pipeline.Germline variant caller uses SNPs,Indels,CNVs and SVs as well.

For the fifth step, reads are aligned using BWA mem and are then sorted by the same algorithm .The reads are then indexed and a VCF file is produced using DeepVariantThe variants are filtered and annotated using the tools like Snpedd,snpsift,annovar,Frequency db,gnomad, dbsnp and RareVariantVis.

Finally, the data is visualised using tools that include GV ,UCSC and RareVariantVis.The plots are available in heatmaps,circos plots,scatter plots, log2 plots,etc

## References
